app/test/unit/git/stash-test.ts

```diff
@@ -0,0 +1,29 @@
+import * as FSE from 'fs-extra'
+import * as path from 'path'
+import { Repository } from '../../../src/models/repository'
+import { setupEmptyRepository } from '../../helpers/repositories'
+import { GitProcess } from 'dugite'
+import { MagicStashString, getStashEntries } from '../../../src/lib/git/stash'
+
+describe('git/stash', () => {
+  describe('getStashEntries', () => {
+    it('returns all stash entries created by desktop', async () => {
+      const repository = await setupEmptyRepository()
+
+      await FSE.writeFile(path.join(repository.path, 'README.md'), 'Hi world\n')
+      await stash(repository)
+
+      const stashEntries = await getStashEntries(repository)
```

Comment:
You can't stash inside an unborn repository:

```shellsession
$ git init blank-repo
Initialized empty Git repository in /Users/shiftkey/src/test/blank-repo/.git/
$ cd blank-repo
$ touch FOO.md
$ git stash push -m "something"
fatal: bad revision 'HEAD'
fatal: bad revision 'HEAD'
fatal: Needed a single revision
You do not have the initial commit yet
```

Reply:
Oops

> You do not have the initial commit yet

☝️ (I did this in my repo I'm testing manually, but forgot to do in test 🤦‍♂️ )